x-pack/elastic-agent/pkg/agent/cmd/container.go

```diff
@@ -437,6 +456,78 @@ func truncateString(b []byte) string {
 	return strings.Replace(string(runes), "\n", " ", -1)
 }
 
+//runLegacyAPMServer by extracting bundled apm-server from elastic-agent
+//and passing on args
+func runLegacyAPMServer(log *logp.Logger, path string, args []string) error {
+	name := "apm-server"
+	log.Info("Preparing apm-server for legacy mode.")
+	log.Infof("Extracting apm-server into install directory %s.", path)
+	cfg := artifact.DefaultConfig()
+
+	installer, err := tar.NewInstaller(cfg)
```

Comment:
@simitt Will this extraction happen on runtime? Do we have sufficient permissions for this in the Cloud container?

Reply:
Yes, that happens during runtime. As long as the docker user has write access to the `APM_SERVER_PATH` this shouldn't be a problem. The path is passed in from the cloud, so I expect this to work.